pkg/controller/pxc/users.go

```diff
@@ -0,0 +1,152 @@
+			Name:      os.Getenv("HOSTNAME"),
+		},
+		&operator,
+	)
+	if err != nil {
+		return errors.Wrap(err, "get operator deployment")
+	}
+
+	for _, secretName := range cr.Spec.Users.Secrets {
+		err = r.handleUsersSecret(secretName, operator, secretObj, cr)
+		if err != nil {
+			log.Error(err, "handle users secret "+secretName)
+		}
+	}
+
+	return nil
+}
+
+func (r *ReconcilePerconaXtraDBCluster) handleUsersSecret(secretName string, operatorPod corev1.Pod, secretObj corev1.Secret, cr *api.PerconaXtraDBCluster) error {
+	containerImage := operatorPod.Spec.Containers[0].Image
+	imagePullSecrets := operatorPod.Spec.ImagePullSecrets
+	usersSecretObj := corev1.Secret{}
+	err := r.client.Get(context.TODO(),
+		types.NamespacedName{
+			Namespace: cr.Namespace,
+			Name:      secretName,
+		},
+		&usersSecretObj,
+	)
+	if err != nil && k8serrors.IsNotFound(err) {
+		return nil
+	} else if err != nil {
+		return errors.Wrapf(err, "get users secret '%s'", "secret-for-users")
+	}
+
+	newHash := ""
+	if secretData, ok := usersSecretObj.Data["secret.yaml"]; ok {
+		newHash = sha256Hash(secretData)
+	}
+	lastAppliedHash := ""
+	if hash, ok := usersSecretObj.Annotations["last-applied"]; ok {
+		lastAppliedHash = hash
+		if lastAppliedHash == newHash {
+			if usersSecretObj.Annotations["status"] == "succeded" || usersSecretObj.Annotations["status"] == "failed" {
+				return nil
+			}
+		}
+	}
+
+	if len(usersSecretObj.Annotations) == 0 {
+		usersSecretObj.Annotations = make(map[string]string)
+	}
+	usersSecretObj.Annotations["last-applied"] = newHash
+	err = r.client.Update(context.TODO(), &usersSecretObj)
+	if err != nil {
+		return errors.Wrap(err, "update secret last-applied")
+	}
+
+	job := users.Job(cr)
+	job.Spec = users.JobSpec(string(secretObj.Data["root"]), cr.Name+"-pxc", containerImage, job, cr, imagePullSecrets)
+
+	currentJob := new(batchv1.Job)
+
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: job.Name, Namespace: cr.Namespace}, currentJob)
+	if err != nil && k8serrors.IsNotFound(err) {
+		err = r.client.Create(context.TODO(), job)
+		if err != nil {
+			return errors.Wrapf(err, "create job '%s'", job.Name)
+		}
+		if len(lastAppliedHash) > 0 {
+			usersSecretObj.Annotations["status"] = "applying"
+			err = r.client.Update(context.TODO(), &usersSecretObj)
+			if err != nil {
+				return errors.Wrap(err, "update secret status")
+			}
+		}
+		time.Sleep(time.Second * 5)
+		return nil
+	} else if err != nil {
+		return errors.Errorf("get user manager job '%s': %v", job.Name, err)
+	}
+
+	if currentJob.Status.Succeeded+currentJob.Status.Failed > 0 {
+		status := ""
+		if currentJob.Status.Succeeded > 0 {
+			status = "succeded"
+		}
+		if currentJob.Status.Failed > 0 {
+			status = "failed"
+		}
+		usersSecretObj.Annotations["status"] = status
+		err = r.client.Update(context.TODO(), &usersSecretObj)
+		if err != nil {
+			return errors.Wrap(err, "update secret status")
+		}
+		err = r.client.Delete(context.TODO(), currentJob)
```

Comment:
how it is possible to debug if a failed job deleted

Reply:
done https://github.com/percona/percona-xtradb-cluster-operator/pull/342/commits/3417c02c2c6246ae92274a0c62f72501f6937a3b